### PR TITLE
dialects: (accfg) Update ACCFG dialect to match version in snax-mlir

### PR DESCRIPTION
**Pull request by AntonLydike**
The `accfg` dialect in snax-mlir has diverged slightly from what it is in upstream xDSL. This PR aims to change that so that we can start upstreaming passes from snax-mlir as well.

Some of things that were reconverged:
* Introduction of `launch_fields` instead of a single `launch_addr`
* Introduction of accfg-effect to model side-effects.
* Introduction of custom syntax on the setup op for improved legibility

### CODE REVIEW

File: xdsl/dialects/accfg.py [hunk 1]
```diff
@@ -303,16 +281,180 @@ def verify_(self) -> None:
...
+
+    def verify_(self) -> None:
+        for _, val in self.fields.data.items():
+            if not isinstance(val, IntegerAttr):
+                raise VerifyException("fields must only contain IntegerAttr!")
+
+    def field_names(self) -> tuple[str, ...]:
+        return tuple(self.fields.data.keys())
+
+    def field_items(self) -> Iterable[tuple[str, IntegerAttr]]:
```
Comment (superlopuh) [inline xdsl/dialects/accfg.py:421]:
```suggestion
    def field_items(self) -> Iterable[tuple[str, IntegerAttr[Attribute]]]:
```

File: xdsl/dialects/accfg.py [hunk 2]
```diff
@@ -72,117 +105,58 @@ def __init__(self, accelerator: str | StringAttr):
...
     """
 
-    def get_sym_attr_name(self, op: Operation) -> StringAttr | None:
-        assert isinstance(op, AcceleratorOp)
-        return StringAttr(op.name_prop.string_value())
-
+    name = "accfg.launch"
 
-@irdl_op_definition
-class AcceleratorOp(IRDLOperation):
```
Comment (AntonLydike) [inline xdsl/dialects/accfg.py:86]: This op was moved further down the document, sadly the diff isn't very clear here. Can re-arrange if needed

File: xdsl/dialects/accfg.py (hunk 1 above)
Comment (AntonLydike) [reply xdsl/dialects/accfg.py]: Doesn't work :(

Do you know what's the best way here?

Comment (AntonLydike) [reply xdsl/dialects/accfg.py]: Ah, found a fix. Apparently `AnyIntegerAttr` just works

**Review by superlopuh — APPROVED** [review]

**Comment by JosseVanDelm** [comment]
Hi @AntonLydike I edited your description a bit thanks for your work!

**Review by JosseVanDelm — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**